core/registration.c

```diff
@@ -857,10 +857,13 @@ static void prv_handleRegistrationUpdateReply(lwm2m_context_t * contextP,
             LOG_ARG_DBG("%d Registration update failed", dataP->server->shortID);
         }
     }
-    if (packet != NULL && packet->code != COAP_231_CONTINUE)
-    {
-        lwm2m_free(dataP->payload);
-        transaction_free_userData(contextP, transacP);
+    if (packet != NULL && packet->code == COAP_231_CONTINUE) {
```

Comment:
If I understand your code correctly, you want to free the user data payload only if `transaction_free_userData` was able to find the actual transaction in the global list and was able to free the user data. Correct me if I'm wrong, but if the transaction cannot be found anymore it should have been freed correctly before. So this might prevent a double free in this case?

Why not inverting the logic to get rid of the empty block after the if-statement?
```suggestion
    if (packet != NULL && packet->code != COAP_231_CONTINUE) {
        uint8_t *payload = dataP->payload;
        if (transaction_free_userData(contextP, transacP)) {
            lwm2m_free(payload);
        }
    }
```

Also can you rebase your change instead of merging? This would remove the merge commit from your branch.

Reply:
Your understanding is correct, but that's not the point.

I reversed the logic to still be able to free userdata when packet is NULL.

Reply:
The key is that when packet==NULL, my code and the original code execute different logic.

Reply:
https://github.com/eclipse-wakaama/wakaama/pull/867/commits
this is a new pull request with a rebase